core/src/main/java/com/linecorp/armeria/client/limit/DefaultConcurrencyLimit.java

```diff
@@ -65,7 +75,24 @@ int acquiredPermits() {
 
     @VisibleForTesting
     int availablePermits() {
-        return maxConcurrency - acquiredPermits.get();
+        int availablePermitCount = maxConcurrency() - acquiredPermits.get();
+        return Math.max(availablePermitCount, 0);
+    }
+
+    @VisibleForTesting
+    int maxConcurrency() {
+        Integer maxConcurrency = this.maxConcurrency.get();
+        if (maxConcurrency == null) {
+            logger.debug("maxConcurrency must not be null. All requests will be pending.");
+            return 0;
+        }
+
+        if (maxConcurrency <= 0) {
```

Comment:
I guess it can return 0 when users want to block the request temporarily.
```suggestion
        if (maxConcurrency < 0) {
```